src/main/java/io/neonbee/NeonBee.java

```diff
@@ -260,12 +256,17 @@ static Future<NeonBee> create(Function<VertxOptions, Future<Vertx>> vertxFactory
         CompositeMeterRegistry compositeMeterRegistry = new CompositeMeterRegistry();
         vertxOptions.setMetricsOptions(
                 new MicrometerMetricsOptions().setMicrometerRegistry(compositeMeterRegistry).setEnabled(true));
-        if (options.isClustered()) {
-            vertxOptions.setClusterManager(clusterManagerFactory.apply(options));
-        }
+
+        Future<VertxOptions> loadClusterManager = Future.future(p -> {
+            if (options.isClustered()) {
+                p.handle(clusterManagerFactory.apply(options).map(vertxOptions::setClusterManager));
+            } else {
+                p.complete(vertxOptions);
+            }
+        });
```

Comment:
Just a thought: we could remove this if block by providing a NO_CLUSTER_FACTORY ClusterManagerFactory that simply returns null. But then we have to make this decision in the NeonBeeOptions.

Reply:
We will not do this as discussed